### ISSUE TITLE
Can't connect to Cassandra 4.0-beta4

### ISSUE THREAD

**Issue by Jorgvt** [HadesArchitect/GrafanaCassandraDatasource#83]
Hi! 
First of all, thanks for your effort creating this plugin.

Currently I have a working Cassandra Cluster being monitored with Prometheus and Grafana and wanted to try your plugin. I installed it following the instructions but I can't make it connect with Cassandra. Do I need to set up anything else in Cassandra? I can normally connect with cqlsh and haven't had problems trying out Presto (so I assume the connection can be stablished).

When doing `sudo tail /var/log/grafana/grafana.log` I can't see anything suspicious, it only says `"Request Completed"`.

I'm using the lastest Grafana available (7.4.2).

**Comment by HadesArchitect** [comment]
Hi @Jorgvt, thank you!

To simplify the initial setup, I've prepared a [quick demo](https://github.com/HadesArchitect/GrafanaCassandraDatasource/wiki/Quick-Demo), could you try it and check if it works for you?

Regarding your question, usually, no additional Cassandra setup required. You can enable more detailed logging by setting Grafana's loglevel to debug, f.e. (if you are using docker)

```
services:
  grafana:
    image: grafana/grafana:7.3.6
    ports:
      - "3000:3000"
    environment:
      GF_LOG_LEVEL: debug  <--------  THIS
``` 

So three more questions here:

1. Operating System
2. Docker or local installation
3. **What exactly happens?** Does the error happen on adding the datasource? After it? What's the error message in grafana web interface?

**Comment by Jorgvt** [comment]
Thanks for the quick response!

I can't currently try out the docker demo because we don't have it set up, but if it's needed I can try to config everything and try it. As of the other questions:

1. We're using Ubuntu 20.04 in an AWS server.
2. Using local installation.
3. As far as I can tell, I can add the plugin easily to Grafana. Then, when I try to create the DataSource, I specify the IP:Port, username and password and click `Save & Test`. Then it says `Unable to establish connection with the database`.

It's weird because every other tool we tryed could connect with the same parameters

**Comment by HadesArchitect** [comment]
Ok I see. Thank you for the details. Which version do you use, 0.4.2? Could you paste the logs of the plugin? The grafana logs I mean.

**Comment by Jorgvt** [comment]
Yea, currently using 0.4.2
[grafana.log](https://github.com/HadesArchitect/GrafanaCassandraDatasource/files/6022807/grafana.log)

I don't want to bother you a lot so if it's not something very obvious don't worry, I'll keep trying hahaha

**Comment by HadesArchitect** [comment]
No worries. You have debug level `info` but we need `debug`. Please refer to grafana docs to raise debug level (should be something easy like an env var)

**Comment by Jorgvt** [comment]
Whoops! Sorry! Still learning a lot... This should have everything you need.
[grafana.log](https://github.com/HadesArchitect/GrafanaCassandraDatasource/files/6022895/grafana.log)

**Comment by HadesArchitect** [comment]
Hmmm that's interesting...

```
msg="2021-02-22T15:12:19.450Z [DEBUG] cassandra-backend-datasource: Connecting to 172.31.8.226:9042..." logger=plugins.backend pluginId=hadesarchitect-cassandra-datasource
t=2021-02-22T15:12:19+0000 lvl=dbug msg= logger=plugins.backend pluginId=hadesarchitect-cassandra-datasource
t=2021-02-22T15:12:19+0000 lvl=dbug msg="2021/02/22 15:12:19 gocql: unable to dial control conn 172.31.8.226:9042: gocql: received unexpected frame on stream 0: [header version=[version=4 direction=RESP] flags=0x0 stream=0 op=SUPPORTED length=96]" logger=plugins.backend pluginId=hadesarchitect-cassandra-datasource
t=2021-02-22T15:12:19+0000 lvl=dbug msg="2021-02-22T15:12:19.454Z [ERROR] cassandra-backend-datasource: Unable to establish connection with the database, gocql: unable to create session: control: unable to connect to initial hosts: gocql: received unexpected frame on stream 0: [header version=[version=4 direction=RESP] flags=0x0 stream=0 op=SUPPORTED length=96]" logger=plugins.backend pluginId=hadesarchitect-cassandra-datasource
```

@futuarmo Any ideas?

**Comment by futuarmo** [comment]
Hello! Please, provide us the version of installed Cassandra

**Comment by Jorgvt** [comment]
Hi! 
This is what pops up when doing `show VERSION`:
`[cqlsh 5.0.1 | Cassandra 4.0-beta4 | CQL spec 3.4.5 | Native protocol v4]`

I think is the lastest available.

**Comment by HadesArchitect** [comment]
OK that makes sense. Thank you for the Bug report!

**Comment by Jorgvt** [comment]
Thanks to you for helping me out!
Are you aware of another versions that might work properly? I will try Cassandra 3.x tomorrow just in case

**Comment by HadesArchitect** [comment]
3.11 works perfectly. Well, actually 4.0-beta1 did well too! 😅

**Comment by Jorgvt** [comment]
I can confirm that Cassandra 3.11.10 works perfectly :P
Thank you very much for everything!

**Comment by HadesArchitect** [comment]
Great to hear. I'll keep this ticket open as we still have to fix the issue.

**Comment by sanchitmehra** [comment]
Hi @HadesArchitect , 
I saw your demo of adding cassandra as datasource with grafana. 
I tried it but I couldn't make it work. 
I am using Windows Machine. Just wanted to make sure if that works on a Window's machine?

**Comment by futuarmo** [comment]
Hello @sanchitmehra. Can you please provide us logs of grafana and maybe describe errors or problems?

**Comment by sanchitmehra** [comment]
> Hello @sanchitmehra. Can you please provide us logs of grafana and maybe describe errors or problems?

Sure. Let me see how to get logs. 
The problem is that even after following the steps you mentioned to add cassandra datasource, I am not able to see cassandra in the list of datasources in grafana. 
Can you please tell what can be the issue. Could it be because of windows Machine?

**Comment by futuarmo** [comment]
Yes, it could be because of windows machine. We didn't test it. Also we have [issue](https://github.com/HadesArchitect/GrafanaCassandraDatasource/issues/59) for that. Thank you for feedback, we'll test in the near future

**Comment by HadesArchitect** [comment]
Hi @sanchitmehra 

1. This ticket relates to Cassandra 4.0, your issue is different. Please create a new issue. 
2. To get the answer, we need logs. Please provide logs by using `docker-compose logs grafana` command

**Comment by futuarmo** [comment]
It's bug in Cassandra 4.0-beta4, fixed in Cassandra 4.0-beta5. We don't need to make any changes
https://issues.apache.org/jira/browse/CASSANDRA-16424?focusedCommentId=17279621&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-17279621

**Comment by HadesArchitect** [comment]
Sweeeeeet!
